src/Routing/Filter/CacheFilter.php

```diff
@@ -106,24 +107,26 @@ public function beforeDispatch(Event $event) {
 	 *
 	 * @return string|null
 	 */
-	public function getFile($url, $mustExist = true) {
+	public function getFile($url, $keyGenerator) {
 		if ($url === '/') {
 			$url = '_root';
 		}
 
+		if($keyGenerator){
+			$url = $keyGenerator($url);
+		}
+
 		$path = $url;
 		$prefix = Configure::read('Cache.prefix');
 		if ($prefix) {
 			$path = $prefix . '_' . $path;
 		}
 
-		if ($url !== '_root') {
-			$path = Inflector::slug($path);
-		}
+		$path = Inflector::slug($path);
```

Comment:
Why is root handled now differently?

Reply:
It can not be assumed that the root url will be '_root' after it has passed through the keyGenerator.

Reply:
I guess we will have to add more tests for both Master and 3.x.

Reply:
Actually, I think you are operating a bit too biased from your use case.
The _root handling here is still valid, as this part is only triggered by the "default case", no?
Once you have a custom callable, it wouldnt try to check on this anyway.

Reply:
No problem. I have reverted the handling of root to how it was before.